src/main/java/org/eclipse/yasson/internal/model/PropertyModel.java

```diff
@@ -457,16 +496,157 @@ public String getWriteName() {
      * Since this is cached for performance reasons strategy has to be consistent
      * with calculated values for same input.
      */
-    private String calculateReadWriteName(String readWriteName, PropertyNamingStrategy strategy) {
+    private static String calculateReadWriteName(String readWriteName, String propertyName, PropertyNamingStrategy strategy) {
         return readWriteName != null ? readWriteName : strategy.translateName(propertyName);
     }
+    
+    /**
+     * Field of a javabean property.
+     *
+     * @return {@link Field field}
+     */
+    public Field getField() {
+        return field;
+    }
 
     /**
-     * Wrapper object of {@code java.lang.reflect} representations of this javabean property.
+     * Setter of a javabean property.
      *
-     * @return Property model
+     * @return {@link Method getter}
      */
-    public PropertyValuePropagation getPropagation() {
-        return propagation;
+    public Method getGetter() {
+        return getter;
+    }
+
+    /**
+     * Getter of a javabean property.
+     *
+     * @return {@link Method setter}
+     */
+    public Method getSetter() {
+        return setter;
+    }
+    
+    // Used in ClassParser
+    public static boolean isPropertyReadable(Field field, Method getter, PropertyVisibilityStrategy strategy) {
+        return createReadHandle(field, getter, isMethodVisible(getter, strategy), strategy) != null;
+    }
+    
+    private static MethodHandle createReadHandle(Field field, Method getter, boolean getterVisible, PropertyVisibilityStrategy strategy) {
+        boolean fieldReadable = field == null || (field.getModifiers() & (Modifier.TRANSIENT | Modifier.STATIC)) == 0;
```

Comment:
If `field == null` then `fieldReadable` will be true, which doesn't seem right

Reply:
I don't think that field can be null there. There's no way I think. I think I kept it here, because the original code had it like too... (code from PropertyValuePropagation)
```
private void initReadable(Field field, Method getter) {
        final boolean fieldReadable = field == null || (field.getModifiers() & (Modifier.TRANSIENT | Modifier.STATIC)) == 0;
        if (!fieldReadable) {
            readable = false;
            return;
        }
        if (getter != null && getterVisible) {
            acceptMethod(getter, OperationMode.GET);
            readable = true;
        } else if (isFieldVisible(field, getter)) {
            acceptField(field, OperationMode.GET);
            readable = true;
        }
    }
```

I think the null check part is redundant